test/integration/getEntry.test.ts

```diff
@@ -14,74 +14,93 @@ const localeClient = contentful.createClient(localeSpaceParams)
-      [{ sys: { id: 'happycat', linkType: 'Entry', type: 'Link' } }, true],
-    ],
-    [
-      'client.withAllLocales',
-      'with all locales, with link resolution',
-      clientWithAllLocales,
-      { 'en-US': 'rainbow' },
-      [{ 'en-US': { sys: { id: 'happycat', linkType: 'Entry', type: 'Link' } } }, true],
-    ],
-    [
-      'client.withoutLinkResolution',
-      'without link resolution',
-      clientWithoutLinkResolution,
-      'rainbow',
-      [{ sys: { id: 'happycat', linkType: 'Entry', type: 'Link' } }, false],
-    ],
-    [
-      'client.withAllLocales.withoutLinkResolution',
-      'with all locales, without link resolution',
-      clientWithAllLocalesAndWithoutLinkResolution,
-      { 'en-US': 'rainbow' },
-      [{ 'en-US': { sys: { id: 'happycat', linkType: 'Entry', type: 'Link' } } }, false],
-    ],
-    [
-      'client.withoutLinkResolution.withAllLocales',
-      'with all locales, without link resolution',
-      clientWithoutLinkResolutionAndWithoutLinkResolution,
-      { 'en-US': 'rainbow' },
-      [{ 'en-US': { sys: { id: 'happycat', linkType: 'Entry', type: 'Link' } } }, false],
-    ],
-  ]
-
-  test.each(matrix)(
-    'getEntry: %s (%s)',
-    // @ts-ignore
-    async (
-      _name,
-      _description,
-      client,
-      expectedLocalization,
-      // @ts-ignore
-      [unresolvedLink, shouldBeResolved]
-    ) => {
-      // @ts-ignore
-      const entry = await client.getEntry(entryId)
-
-      // checking for expected localization
-      expect(entry.fields.color).toEqual(expectedLocalization)
-
-      // checking for expected entry link resolution
-      shouldBeResolved
-        ? expect(entry.fields.bestFriend).not.toEqual(unresolvedLink)
-        : expect(entry.fields.bestFriend).toEqual(unresolvedLink)
-    }
-  )
+  const entryWithUnresolvableLink = '4SEhTg8sYJ1H3wDAinzhTp'
+  const entryWithResolvableLink = 'nyancat'
+
+  describe('default client', () => {
+    test('client', async () => {
+      const response = await client.getEntry(entryWithResolvableLink, {
+        include: 2,
+      })
+
+      expect(response.fields.bestFriend.sys.type).toBe('Entry')
+      expect(response.fields.color).toBe('rainbow')
+      expect(response.fields.color['en-US']).not.toBeDefined()
+    })
+  })
+
+  describe('client has withoutUnresolvableLinks modifier', () => {
+    test('client.withoutUnresolvableLinks', async () => {
+      const response = await client.withoutUnresolvableLinks.getEntry(entryWithUnresolvableLink, {
+        include: 2,
+      })
+
+      expect(response.fields.bestFriend).toBeUndefined()
+    })
+
+    test.skip('client.withoutUnresolvableLinks.withAllLocales', async () => {
+      const response = await client.withoutUnresolvableLinks.withAllLocales.getEntry(
+        entryWithUnresolvableLink,
+        {
+          include: 2,
+        }
+      )
+
+      expect(response.fields.color).toHaveProperty('en-US')
+      //TODO: why are we returning an empty object?
```

Comment:
Because `withoutUnresolvableLinks` returns different things, depending on `withAllLocales` or not:
```js
client.withoutUnresolvableLinks.getEntries() // returns no `bestFriend` property, hence it's undefined.
client.withAllLocales.withoutUnresolvableLinks.getEntries() // returns `bestFriend: {}`
```
(this is nothing we changed, it matches the current behavior)

Reply:
Yes, we didn't change the current behavior, the question is more for the future, is it something that we want to keep working like that? because there is no advantage of using  withoutUnresolvableLinks with locale 🤔